azure-mgmt-search/src/main/java/com/microsoft/azure/management/search/SearchService.java

```diff
@@ -0,0 +1,319 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+package com.microsoft.azure.management.search;
+
+import com.microsoft.azure.CloudException;
+import com.microsoft.azure.management.apigeneration.Beta;
+import com.microsoft.azure.management.apigeneration.Fluent;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.GroupableResource;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.Resource;
+import com.microsoft.azure.management.resources.fluentcore.model.Appliable;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
+import com.microsoft.azure.management.resources.fluentcore.model.Refreshable;
+import com.microsoft.azure.management.resources.fluentcore.model.Updatable;
+import com.microsoft.azure.management.search.implementation.SearchServiceInner;
+import com.microsoft.azure.management.search.implementation.SearchServiceManager;
+import rx.Completable;
+import rx.Observable;
+
+import java.util.List;
+
+/**
+ * An immutable client-side representation of an Azure registry.
+ */
+@Fluent
+@Beta(Beta.SinceVersion.V1_2_0)
+public interface SearchService extends
+    GroupableResource<SearchServiceManager, SearchServiceInner>,
+    Refreshable<SearchService>,
+    Updatable<SearchService.Update> {
+
+  /***********************************************************
+   * Getters
+   ***********************************************************/
+
+  /**
+   * Get the hostingMode value.
```

Comment:
@martinsawicki Shouldn't AutoRest implement these rules to make this a less manual process?